### PR TITLE
chore: Build rust-peer for arm

### PR DESCRIPTION
**Pull request by nahsi**

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,3 +1,10 @@
 [toolchain]
 channel = "nightly-2023-02-27"
-components = ["rustfmt", "clippy"]
+components = [
+  "rustfmt",
+  "clippy"
+]
+targets = [
```
Comment (gurinderu) [inline rust-toolchain.toml:7]: not needed

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -47,12 +49,15 @@ jobs:
         flavour:
           - ipfs
           - rich
+        include:
+          - arch: x86_64
+          # - arch: aarch64
```
Comment (gurinderu) [inline .github/workflows/e2e.yml:54]: remove this line, is it not needed in e2e

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -24,10 +24,20 @@ jobs:
 
     env:
       RUSTFLAGS: "-D warnings --cfg tokio_unstable"
+      CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER: aarch64-linux-gnu-gcc
```
Comment (gurinderu) [inline .github/workflows/build.yml:27]: this envs is a platform-specific,  probably need to declare them inside matrix section ?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -24,10 +24,20 @@ jobs:
...
 
     outputs:
       sha256: "${{ steps.sha.outputs.sha256 }}"
 
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch: x86_64
+          # - arch: aarch64
```
Comment (gurinderu) [inline .github/workflows/build.yml:39]: uncomment?)

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -18,16 +18,29 @@ on:
 
 jobs:
   build:
-    name: "Build rust-peer"
+    name: "Build rust-peer (${{ matrix.arch }})"
```
Comment (folex) [inline .github/workflows/build.yml:21]:
```suggestion
    name: "Build nox (${{ matrix.arch }})"
```

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -71,20 +84,10 @@ jobs:
...
-          sha=($(sha256sum particle-node))
-          echo "sha256=${sha}" >> $GITHUB_OUTPUT
+        run: cargo build ${{ steps.profile.outputs.flags }} -p particle-node --target ${{ matrix.arch }}-unknown-linux-gnu
 
       - name: Upload rust-peer binary
         uses: actions/upload-artifact@v3
         with:
-          name: rust-peer
-          path: target/${{ steps.profile.outputs.profile }}/particle-node
+          name: rust-peer-${{ matrix.arch }}
```
Comment (folex) [inline .github/workflows/build.yml:92]:
```suggestion
          name: nox-${{ matrix.arch }}
```

**Pull request merged**